modules/local/merge_predictions/templates/merge_predictions.py

```diff
@@ -0,0 +1,308 @@
+        return 10 ** ((1 - BA) * math.log10(50000))
+
+    @staticmethod
+    def longTowide(df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms a long-format DataFrame into a wide-format DataFrame,
+        where 'predictor-allele-BA', 'predictor-allele-rank', and 'predictor-allele-binder'
+        become separate columns.
+
+        Parameters:
+            df (pd.DataFrame): The original long-format DataFrame.
+
+        Returns:
+            pd.DataFrame: Transformed wide-format DataFrame.
+        """
+        # Identify non-predictor columns to keep as index
+        meta_columns = [col for col in df.columns if col not in ['predictor', 'allele', 'BA', 'rank', 'binder']]
+
+        # Pivot to wide format
+        df_wide = df.pivot_table(
+            index=meta_columns,
+            columns=['predictor', 'allele'],
+            values=['BA', 'rank', 'binder'],
+            aggfunc='first'  # Assuming first occurrence is enough if duplicates exist
+        )
+
+        # Flatten the MultiIndex columns
+        df_wide.columns = [f"{pred}_{allele}_{val}" for val, pred, allele in df_wide.columns]
+
+        # Reset index
+        df_wide.reset_index(inplace=True)
+
+        return df_wide
+
+
+class PredictionResult:
+    def __init__(self, file_path, alleles, peptide_col_name):
+        self.file_path = file_path
+        self.alleles = alleles
+        self.peptide_col_name = peptide_col_name
+        self.predictor = None
+        self.prediction_df = self._format_prediction_result()
+
+    def _format_prediction_result(self):
+        """
+        Returns a harmonized DataFrame with the prediction results. Output format:
+        +----------+---------+-------+-------+--------+-----------+
+        | sequence | allele  | BA    | rank  | binder | predictor |
+        +----------+---------+-------+-------+--------+-----------+
+        |    ...   |   ...   |  ...  |  ...  |  ...   |     ...   |
+        +----------+---------+-------+-------+--------+-----------+
+        """
+        if 'syfpeithi' in self.file_path:
+            self.predictor = 'syfpeithi'
+            return self._format_syfpeithi_prediction()
+        elif 'mhcflurry' in self.file_path:
+            self.predictor = 'mhcflurry'
+            return self._format_mhcflurry_prediction()
+        elif 'mhcnuggets' in self.file_path:
+            self.predictor = 'mhcnuggetsii' if 'mhcnuggetsii' in self.file_path else 'mhcnuggets'
+            return self._format_mhcnuggets_prediction()
+        elif 'netmhcpan' in self.file_path:
+            self.predictor = 'netmhcpan'
+            return self._format_netmhcpan_prediction()
+        elif 'netmhciipan' in self.file_path:
+            self.predictor = 'netmhciipan'
+            return self._format_netmhciipan_prediction()
+        else:
+            logging.error(f'Unsupported predictor type in file: {self.file_path}.')
+            sys.exit(1)
+
+    def _format_syfpeithi_prediction(self):
```

Comment:
What is the purpose of the empty def?

Reply:
Initially was there as a placeholder :P , but decided to remove syfpeithi

Reply:
Why removing SYFPEITHI?

Reply:
Legacy predictor, complicated supported lenghts and alleles, almost no-one using it (also not according to nf-core slack, but hardly anyone voted)

Reply:
I'd rather include MixMHCpred (if possible) or modern predictors

Reply:
I get that tbh 😃